src/Controls/src/Core/Handlers/Shell/Windows/ShellView.cs

```diff
@@ -123,19 +123,31 @@ internal void UpdateMenuItemSource()
 				_flyoutGrouping = newGrouping;
 				var newItems = IterateItems(newGrouping).ToList();
 
-				foreach (var item in newItems)
+				for (int index = 0; index < newItems.Count; index++)
 				{
-					if (!FlyoutItems.Contains(item))
+					var item = newItems[index];
+					var flyoutItemIndex = FlyoutItems.IndexOf(item);
+
+					if (flyoutItemIndex == -1)
```

Comment:
Not big performance differences, but can simplify this using `Contains`:

```
if (!FlyoutItems.Contains(item))
    {
        // Use Insert when within bounds, otherwise Add
        if (index < count)
        {
            FlyoutItems.Insert(index, item);
        }
        else
        {
            FlyoutItems.Add(item);
        }
    }
```

NOTE: count is a variable with the `FlyoutItems.Count `value.